js/index_controls.js

```diff
@@ -116,35 +107,29 @@ function calculate() {
+					maxHealthRecovered = Math.floor(p1.maxHP * p2.moves[i].percentHealed * 1000 / p2.maxHP) / 10
 				} else if (notation !== '%' && minHealthRecovered > 48) {
-					minHealthRecovered = 48;
-					maxHealthRecovered = 48;
+					minHealthRecovered = Math.floor(p1.maxHP * p2.moves[i].percentHealed * 48 / p2.maxHP);
+					maxHealthRecovered = Math.floor(p1.maxHP * p2.moves[i].percentHealed * 48 / p2.maxHP);
 				}
 			recoveryText = ' (' + minHealthRecovered + ' - ' + maxHealthRecovered + notation + ' recovered)';
 		}
 		var recoilText = '';
-		if (p2.moves[i].hasRecoil && hasAThirdRecoil) {
-			var minRecoilDamage = notation === '%' ? Math.floor(minDamage * (1/3) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/3) * 48 / p2.maxHP);
-			var maxRecoilDamage = notation === '%' ? Math.floor(maxDamage * (1/3) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/3) * 48 / p2.maxHP);
-			recoilText = ' (' + minRecoilDamage + ' - ' + maxRecoilDamage + notation + ' recoil damage)';
-		}
-		if (p2.moves[i].hasRecoil && hasAFourthRecoil) {
-			var minRecoilDamage = notation === '%' ? Math.floor(minDamage * (1/4) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/4) * 48 / p2.maxHP);
-			var maxRecoilDamage = notation === '%' ? Math.floor(maxDamage * (1/4) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/4) * 48 / p2.maxHP);
-			recoilText = ' (' + minRecoilDamage + ' - ' + maxRecoilDamage + notation + ' recoil damage)';
-		}
-		if (p2.moves[i].hasRecoil && hasHalfRecoil) {
-			var minRecoilDamage = notation === '%' ? Math.floor(minDamage * (1/2) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/2) * 48 / p2.maxHP);
-			var maxRecoilDamage = notation === '%' ? Math.floor(maxDamage * (1/2) * 1000 / p2.maxHP) / 10 :
-				Math.floor(minDamage * (1/2) * 48 / p2.maxHP);
+		if (typeof p2.moves[i].hasRecoil === 'number') {
+			var minRecoilDamage = notation === '%' ? Math.floor(Math.min(minDamage, p1.maxHP) * p2.moves[i].hasRecoil * 10 / p2.curHP) / 10 :
+				Math.floor(Math.min(minDamage, p1.maxHP) * p2.moves[i].hasRecoil * 0.48 / p2.maxHP);
+			var maxRecoilDamage = notation === '%' ? Math.floor(Math.min(maxDamage, p1.maxHP) * p2.moves[i].hasRecoil * 10 / p2.maxHP) / 10 :
+				Math.floor(Math.min(maxDamage, p1.maxHP) * p2.moves[i].hasRecoil * 0.48 / p2.maxHP);
+			if (minDamage > p1.curHP) {
+				minRecoilDamage = notation === '%' ? Math.floor(Math.min(p1.maxHP * p2.moves[i].hasRecoil) * 10 / p2.maxHP) / 10
+				: Math.floor(p1.maxHP * p2.moves[i].recoilPercentage * 48 / p1.maxHP);
+				maxRecoilDamage = notation === '%' ? Math.floor(Math.min(p1.maxHP, p2.moves[i].hasRecoil) * 10 / p2.maxHP) / 10
+				: Math.floor(Math.min(p1.maxHP, p2.moves[i].hasRecoil) * 0.48 / p2.maxHP);
+			}
 			recoilText = ' (' + minRecoilDamage + ' - ' + maxRecoilDamage + notation + ' recoil damage)';
+		} else if (p2.moves[i].hasRecoil && typeof p2.moves[i].hasRecoil !== 'number') {
```

Comment:
No need to check that it's not a number here, since getting to this point at all means it's not a number. Also keep in mind if you decide to show crash damage text, it's different in almost every gen.

Reply:
I did it like this so every calc doesn’t get recoil text, but checking if hasRecoil check is probably enough

Reply:
(To clarify, checking if it wasn’t a number wasn’t enough as every calc got recoil text at that point)

Reply:
Yeah, I got that, I was only talking about the number check.

You seem to have also added a check for Mind Blown with `hasRecoil` without accounting for it in Reckless calculations (it's not boosted).

Reply:
What other changes are there? I looked at the dex on PS and double edge was the only one?? And oh I forgot about reckless, this is why I didn’t want to mess with hasRecoil :> Also I was talking about doing the else statement that way, but yeah you were right

Reply:
Are you talking about struggle? Struggle isn’t on the calc currently but I am going to do it, just don’t know how much time I’ll have this week and for now this works

Reply:
The text I'm talking about is the one associated with this block, not the one above. I also mentioned it in the first comment here. Bughouse posted the exact specifications for crash damage in each gen in your thread.

Reply:
Oh I haven’t checked smogon since Friday and ah I finally get what you mean. Easy enough fix, I’ll do it when I do struggle. Btw did you say crash damage at first? I read recoil damage so I thought you meant the line above / double edge again, my bad.

Reply:
(You definitely said crash damage, curse adhd I gotta learn to read more thoroughly oml)